scalalib/src/mill/scalalib/CoursierModule.scala

```diff
@@ -108,3 +122,38 @@ trait CoursierModule extends mill.Module {
     T.task { None }
 
 }
+object CoursierModule{
+
+  class Resolver(repositories: Seq[Repository],
+                 bind: Dep => BoundDep,
+                 mapDependencies: Option[Dependency => Dependency] = None,
+                 customizer: Option[coursier.core.Resolution => coursier.core.Resolution] = None,
+                 ctx: Option[mill.api.Ctx.Log] = None,
+                 coursierCacheCustomizer: Option[
+                   coursier.cache.FileCache[coursier.util.Task] => coursier.cache.FileCache[coursier.util.Task]
+                 ] = None) {
+
+    def resolveDeps[T: CoursierModule.Resolvable](deps: IterableOnce[T],
```

Comment:
Using a typeclass here is a bit annoying, but it's necessary to work around the "multiple overloads cant have default params" problem that would otherwise make the user-facing API a lot worse